tests/blockchain/eth/interfaces/test_decorators.py

```diff
@@ -0,0 +1,81 @@
+"""
+This file is part of nucypher.
+
+nucypher is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+nucypher is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import os
+import pytest
+
+from eth_utils import to_checksum_address
+
+from nucypher.blockchain.eth.decorators import validate_checksum_address, InvalidChecksumAddress
+
+
+def get_random_checksum_address():
+    canonical_address = os.urandom(20)
+    checksum_address = to_checksum_address(canonical_address)
+    return checksum_address
+
+
+def test_validate_checksum_address():
+
+    # Simple case: just one parameter, called "checksum_address"
+    @validate_checksum_address
+    def just_one_address(checksum_address):
+        return True
+
+    with pytest.raises(InvalidChecksumAddress):
+        just_one_address("0x_NOT_VALID")
+
+    with pytest.raises(TypeError):
+        just_one_address(123)
+
+    assert just_one_address(get_random_checksum_address())
+
+    # More complex case: the parameter is optional
+    @validate_checksum_address
+    def optional_checksum_address(whatever, staking_address=None):
+        return True
+
+    with pytest.raises(InvalidChecksumAddress):
+        optional_checksum_address(12, "0x_NOT_VALID")
```

Comment:
I guess, a typical wrong one is `get_random_checksum_address().lower()`

Reply:
✅